File: test/data/find_references/refs.go
package main

import "fmt"

func helper() int {
	return 42
}

func process(data string) string {
	result := transform(data)
	helper()
	return result
}

func transform(data string) string {
	return data
}

type Service struct {
	db string
}

func NewService(db string) *Service {
	return &Service{db: db}
}

func (s *Service) Fetch(query string) string {
	return s.db
}

func (s *Service) Run() string {
	data := s.Fetch("SELECT 1")
	return process(data)
}

func main() {
	svc := NewService("db")
	result := svc.Run()
	fmt.Println(result)
}

File: test/data/find_references/refs.rs
fn helper() -> i32 {
    42
}

fn process(data: &str) -> String {
    let result = transform(data);
    helper();
    result
}

fn transform(data: &str) -> String {
    data.to_string()
}

struct Service {
    db: String,
}

impl Service {
    fn new(db: &str) -> Self {
        Service { db: db.to_string() }
    }

    fn fetch(&self, query: &str) -> String {
        self.db.clone()
    }

    fn run(&self) -> String {
        let data = self.fetch("SELECT 1");
        process(&data)
    }
}

fn main() {
    let svc = Service::new("db");
    let result = svc.run();
    println!("{}", result);
}
